test/integration/targets/copy/tasks/dest_in_non_existent_dir.yml

```diff
@@ -0,0 +1,28 @@
+# src is a file, dest is a non-existent directory: checks that dest is created
+- name: Ensure that dest directory doesn't exist
+  file:
+    path: '{{ output_dir }}/new_sub_dir1/'
+    state: absent
+
+- name: Copy file, dest is a non-existing target directory
+  copy:
+    src: '{{ item.src }}'
+    dest: '{{ output_dir }}/new_sub_dir1/{{ item.dest }}'
+  register: copy_result
+
+- name: assert copy worked
+  assert:
+    that:
+      - 'copy_result|success'
+      - 'copy_result|changed'
+
+- name: stat the copied path
+  stat:
+    path: '{{ output_dir }}/new_sub_dir1/sub_dir2/{{ item.check }}'
```

Comment:
I like that all these test cases show two levels of parent directory being created: `new_sub_dir1/` and then `subdir2/`.

That aspect is not necessarily clear when you're looking at the task names / comments.  But you can tell it's deliberate once you find the check for `/new_sub_dir1/sub_dir2/{{ item.check }}`.  So I think it's OK :).